src/components/App/App.js

```diff
@@ -0,0 +1,312 @@
+import SavedNews from "../SavedNews/SavedNews";
+import mainApi from "../../utils/MainApi";
+import { getNewsApi } from "../../utils/NewsApi";
+
+function App() {
+  // AUTH states
+  const [currentUser, setCurrentUser] = useState({});
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [conflictErr, setConflictErr] = useState(false);
+
+  // POPUP visibility states:
+  const [isSigninPopupOpen, setSigninPopupOpen] = useState(false);
+  const [isSignupPopupOpen, setSignupPopupOpen] = useState(false);
+  const [isTooltipOpen, setIsTooltipOpen] = useState(false);
+
+  // NEWS states:
+  const [preloader, setPreloader] = useState(false);
+  const [notFound, setNotFound] = useState(false);
+  const [showArticleSection, setShowArticleSection] = useState(false);
+  const [savedArticles, setSavedArticles] = useState([]);
+  const [searchedArticles, setSearchedArticles] = useState([]);
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  //*** AUTH FUNCTIONS ***=================================
+
+  // keep user logged in when token is set:
+  useEffect(() => {
+    (async () => {
+      try {
+        const user = await mainApi.checkToken(localStorage.getItem("jwt").jwt);
+        if (user) {
+          setCurrentUser({ name: user.name });
+          setLoggedIn(true);
+          mainApi.getUserInfo();
+          console.log(user.name);
+        }
+        const articlesFromDb = await mainApi.getSavedArticles();
+        if (articlesFromDb !== 0) {
+          setSavedArticles(articlesFromDb);
+        }
+        if (location.pathname === "/saved-news") {
+          setShowArticleSection(true);
+        }
+      } catch {
+        return;
+      }
+    })();
+    setSearchedArticles(JSON.parse(localStorage.getItem("articles")) || []);
+  }, [loggedIn, location.pathname]);
+
+  // keep last search results live:
+  useEffect(() => {
+    if (localStorage.getItem("articles")) {
+      setShowArticleSection(true);
+    }
+  }, []);
+
+  //*** USER FUNCTIONS ***=================================
+
+  // signup:
+  async function handleSignupSubmit(data) {
+    try {
+      const userData = await mainApi.signup(data);
+      if (userData) {
+        setIsTooltipOpen(true);
+        // console.log("Sign Up Data:", data);
+        // console.log("Sign Up currentUser is:", currentUser);
+        console.log(userData.name, "you are now a special member");
+        setSignupPopupOpen(false);
+      }
+    } catch (err) {
+      console.log(err, err.response, err.status);
+      if (err) {
+        setConflictErr(true);
+        setTimeout(function () {
+          setConflictErr(false);
+        }, 2000);
+      }
+      setIsTooltipOpen(false);
+      return;
+    }
+  }
+
+  // login:
+  async function handleSigninSubmit(data) {
+    try {
+      //mainApi sets up the "jwt" token:
+      const token = await mainApi.signin(data);
+      if (token) {
+        setLoggedIn(true);
+        window.location.reload();
```

Comment:
`window.location.reload();` please, delete it from your project because React should work without any reloadings. It's the main purpose of SPA-apps